web/pandas/pdeps/0008-inplace-methods-in-pandas.md

```diff
@@ -0,0 +1,403 @@
+    - What should happen when ``inplace=True`` but the original pandas object cannot be operated inplace on because it
+      shares its values with another object?
+
+## Motivation and Scope
+
+The `inplace=True` keyword has been a controversial topic for many years. It is generally seen (at least by several
+pandas maintainers and educators) as bad practice and often unnecessary, but at the same time it is also widely used,
+partly because of confusion around the impact of the keyword.
+
+Generally, we assume that people use the keyword for the following reasons:
+
+1. Because they think it is more efficient (it is faster and/or can save memory)
+2. To save the result to the same variable / update the original variable (avoid the pattern of reassigning to the same
+   variable name)
+
+For the first reason: efficiency is an important aspect. However, in practice it is not always the case
+that `inplace=True` improves anything. Some of the methods with an `inplace` keyword can actually work inplace, but
+others still make a copy under the hood anyway. In addition, with the introduction of Copy-on-Write, there are now other
+ways to avoid making unnecessary copies by default (without needing to specify a keyword). The next section gives a
+detailed overview of those different cases.
+
+For the second reason: we are convinced that this is not worth it. While it might save some keystrokes (if you have a
+long variable name), this code style also has sufficient disadvantages that we think it is not worth providing "two
+ways" to achieve the same result:
+
+- You can't use method chaining with `inplace=True`
+- The ``inplace`` keyword complicates type annotations (because the return value depends on the value of `inplace`)
+- Using `inplace=True` gives code that mutates the state of an object and thus has side-effects. That can introduce
+  subtle bugs and is harder to debug.
+
+Finally, there are also methods that have a `copy` keyword instead of an `inplace` keyword (which also avoids copying
+the data in the case of `copy=False`, but still returns a new object referencing the same data instead of updating the
+calling object), adding to the inconsistencies. This `copy=False` option also has become redundant with the introduction
+of Copy-on-Write.
+
+Given the above reasons, we are convinced that there is no need for neither the `inplace` nor the `copy` keyword (except
+for a small subset of methods that can actually update data inplace). Removing those keywords will give a more
+consistent and less confusing API.
+
+Thus, in this PDEP, we aim to standardize behavior across methods to make control of inplace-ness of operations
+consistent, and compatible with Copy-on-Write.
+
+Note: there are also operations (not methods) that work inplace in pandas, such as indexing (
+e.g. `df.loc[0, "col"] = val`) or inplace operators (e.g. `df += 1`). This is out of scope for this PDEP, as we focus on
+the inplace behaviour of DataFrame and Series _methods_.
```

Comment:
maybe a subsection for out of scope and pull down the out of scope related items from the abstract